oak_utils/src/lib.rs

```diff
@@ -180,27 +198,96 @@ pub fn compile_protos_without_services<P>(inputs: &[P], includes: &[P])
 where
     P: AsRef<std::path::Path>,
 {
-    compile_protos_with(false, inputs, includes);
+    compile_protos_with(&mut prost_build::Config::new(), inputs, includes);
 }
 
-fn compile_protos_with<P>(generate_services: bool, inputs: &[P], includes: &[P])
+fn compile_protos_with<P>(prost_config: &mut prost_build::Config, inputs: &[P], includes: &[P])
 where
     P: AsRef<std::path::Path>,
 {
+    compile_protos_with_internal(
+        prost_config
+            .type_attribute(".", "#[derive(::oak::handle::HandleVisit)]")
+            .extern_path(".oak.handle", "::oak::handle")
+            .extern_path(".oak.encap.GrpcRequest", "::oak::grpc::GrpcRequest")
+            .extern_path(".oak.encap.GrpcResponse", "::oak::grpc::GrpcResponse"),
+        inputs,
+        includes,
+    )
+}
+
+fn compile_protos_with_internal<P>(
+    prost_config: &mut prost_build::Config,
+    inputs: &[P],
+    includes: &[P],
+) where
+    P: AsRef<std::path::Path>,
+{
+    set_protoc_env_if_unset();
     for input in inputs {
         // Tell cargo to rerun this build script if the proto file has changed.
         // https://doc.rust-lang.org/cargo/reference/build-scripts.html#cargorerun-if-changedpath
         println!("cargo:rerun-if-changed={}", input.as_ref().display());
     }
 
-    let mut prost_config = prost_build::Config::new();
-    if generate_services {
-        prost_config.service_generator(Box::new(OakServiceGenerator));
-    }
     prost_config
         // We require label-related types to be comparable and hashable so that they can be used in
         // hash-based collections.
         .type_attribute(".oak.label", "#[derive(Eq, Hash)]")
         .compile_protos(inputs, includes)
         .expect("could not run prost-build");
 }
+
+// For internal use in Oak.
+#[doc(hidden)]
+pub fn compile_protos_to_internal<P, Q>(inputs: &[P], includes: &[P], out_dir: Q)
+where
+    P: AsRef<std::path::Path>,
+    Q: Into<std::path::PathBuf>,
+{
+    compile_protos_with_internal(
+        prost_build::Config::new().out_dir(out_dir),
+        inputs,
+        includes,
+    );
+}
+
+// For internal use in Oak.
+#[doc(hidden)]
+pub fn compile_protos_internal<P>(inputs: &[P], includes: &[P])
+where
+    P: AsRef<std::path::Path>,
+{
+    compile_protos_with_internal(&mut prost_build::Config::new(), inputs, includes);
+}
+
+// For internal use in Oak.
+#[doc(hidden)]
+pub fn compile_protos_internal_with_service_generator<P>(inputs: &[P], includes: &[P])
```

Comment:
nit: Feels slightly odd that the 'normal' versions have a `_without_services` variant, but the `_internal_` versions have a `with_service_generator` variant.  Could we align naming and defaults between the two?